server/server.go

```diff
@@ -2187,7 +2189,7 @@ func findBestQueueSub(sl []*subState) (rsub *subState) {
 
 		// Favor non stalled subscribers and clients that do not have
 		// failed heartbeats
-		if (!sStalled || rStalled) && (!sHasFailedHB || rHasFailedHB) && (sOut < rOut) {
+		if (!sStalled && !sHasFailedHB && (rStalled || rHasFailedHB)) || (sOut < rOut) {
```

Comment:
I have to return a sub, because this function is also used in case of redeliveries where redeliveries are always forced (even if the sub is stalled). If I don't return a sub, test `TestPersistentStoreStalledQueueRedelivery` would fail.